MatrixSDK/JSONModels/Login/MXLoginSSOFlow.m

```diff
@@ -0,0 +1,60 @@
+// 
+// Copyright 2020 The Matrix.org Foundation C.I.C
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#import "MXLoginSSOFlow.h"
+
+NSString *const MXLoginSSOFlowIdentityProvidersKey = @"identity_providers";
+
+@interface MXLoginSSOFlow()
+
+@property (nonatomic, readwrite) NSArray<MXLoginSSOIdentityProvider*> *identityProviders;
+
+@end
+
+@implementation MXLoginSSOFlow
+
++ (id)modelFromJSON:(NSDictionary *)JSONDictionary
+{
+    MXLoginSSOFlow *loginFlow = [MXLoginSSOFlow new];
+    if (loginFlow)
+    {
+        MXJSONModelSetString(loginFlow.type, JSONDictionary[@"type"]);
+        MXJSONModelSetArray(loginFlow.stages, JSONDictionary[@"stages"]);
```

Comment:
Can't we delegate this parsing to the superclass? I mean with another method like `parseJSON:` in `MXLoginfFlow`. My concern is that If another property added to `MXLoginFlow`, we'll have to maintain this one also.

Reply:
Yes he have an architecture problem. Here the class method is not really convenient. Member method like other libs would help to make calls to `super` implementation. 
In `MXLoginSSOFlow` Now we can do something like:
```
+ (instancetype)modelFromJSON:(NSDictionary *)JSONDictionary
{
    MXLoginSSOFlow *loginFlow = [super modelFromJSON:JSONDictionary];
```
And in `MXLoginFlow` we should have: 
```
+ (instancetype)modelFromJSON:(NSDictionary *)JSONDictionary
{
    MXLoginSSOFlow *loginFlow = [self new];
```

Reply:
I think this seems better.